core/commonMain/src/kotlinx/serialization/Serializers.kt

```diff
@@ -18,54 +18,112 @@ import kotlin.reflect.*
 
 /**
  * Retrieves a serializer for the given type [T].
- * This method is a reified version of `serializer(KType)`.
+ * This overload is a reified version of `serializer(KType)`.
+ *
+ * This overload works with full type information, including type parameters and nullability,
+ * and is a recommended way to retrieve a serializer.
+ * For example, `serializer<List<String?>>()` returns [KSerializer] that is able
+ * to serialize and deserialize list of nullable strings — i.e. `ListSerializer(String.serializer().nullable)`.
+ *
+ * Variance of [T]'s type arguments is not significant for serialization.
+ * However, star projections in [T]'s type arguments are not allowed.
+ *
+ * @throws SerializationException if serializer cannot be created (provided [T] or its type argument is not serializable).
+ * @throws IllegalArgumentException if [T] contains star projection.
```

Comment:
```suggestion
 * @throws IllegalArgumentException if any of [T] type arguments is star projection.
```

Reply:
Please apply those to all overloads

Reply:
Probably 'contains star projection is better' because we also may have smth like `List<Box<*>>` where T = List so star projection is not immediate type argument of List

Reply:
"if any of [T] type arguments contains star projection"?

I honestly don't know which one is more precise